src/entities/amm.ts

```diff
@@ -1700,67 +1770,45 @@ class AMM {
   // descale compound tokens
 
   public descaleCompoundValue(value: BigNumber): number {
-    return Number(ethers.utils.formatUnits(value, parseInt(this.underlyingToken.decimals.toString()) + 10));
+    return Number(
+      ethers.utils.formatUnits(value, parseInt(this.underlyingToken.decimals.toString()) + 10),
+    );
   }
 
   // underlying token approval for periphery
 
-  public async isUnderlyingTokenApprovedForPeriphery(): Promise<boolean | void> {
-    if (!this.underlyingToken.id) {
-      throw new Error("No underlying token");
+  // token approval for periphery
+  public async isTokenApprovedForPeriphery(
+    tokenAddress: string | undefined,
```

Comment:
why don't we enforce tokenAddress to be passed (i.e. tokenAddress: string)?

Reply:
Was just maintaining something closer to the existing interface, which threw an error if there was no underlying token. Can move this error throwing outside the function if you prefer but we may then end up with some duplicated code. wdty?

Reply:
I'd keep this typed to string and check before passing to the function.

Reply:
Closing this duplicate conversation.